### PR TITLE
Graph QL: TLU pipeline dsl and remote schema url changes

### PR DESCRIPTION
**Pull request by AleenaThomas-gh**

### CODE REVIEW

**Review by AleenaThomas-gh — COMMENTED** [review]
Merging the pending commit from graphQL.

File: docker-compose.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ services:
     image: timescale/timescaledb:2.5.1-pg14
     restart: unless-stopped
     volumes:
-      - db_data:/var/lib/postgresql/data
+      - db_data1:/var/lib/postgresql/data
```
Comment (fungiboletus) [inline docker-compose.yaml:7]: Why have you renamed the data volume ?

File: docker-compose.yaml [hunk 2]
```diff
@@ -29,6 +29,8 @@ services:
       HASURA_GRAPHQL_ADMIN_SECRET: ${HASURA_ADMIN_SECRET-hasuraadminsecret}
       HASURA_GRAPHQL_UNAUTHORIZED_ROLE: anonymous
       HASURA_GRAPHQL_ENABLE_TELEMETRY: "false"
+      HASURA_GRAPHQL_LOG_LEVEL: debug
+      REMOTE_SCHEMA_URL: http://10.218.149.206:9000
```
Comment (fungiboletus) [inline docker-compose.yaml:33]: Which IP is this one ? Should it be hardcoded ?

File: docker-compose.yaml [hunk 3]
```diff
@@ -41,7 +43,7 @@ services:
       - appsmith:/appsmith-stacks
   
   controller:
-    image: backend-v1
+    image: simpipe-backend-v4
```
Comment (fungiboletus) [inline docker-compose.yaml:46]: You should probably use tags and not the image name to have multiple versions.

Somethink like:

  - `simpipe-backend:v1` and `simpipe-backend:v4`

File: log/controller.log [hunk 4]
```diff
@@ -153,3 +153,394 @@ info🏷️: [Feb-25-2022 08:41:45]: Cleared Sandbox for next simulation
 info🏷️: [Feb-25-2022 08:41:45]: Stored simulation details to simulations/7d1563a6-e015-46b3-a260-3df5db378ae2/6d67e260-92da-4f27-87b6-7024f6efbc8b/3
 info🏷️: [Feb-25-2022 08:41:45]: Step 3 finished execution
 
+info🏷️: [Feb-25-2022 09:40:45]: test1.txt,test 1
```
Comment (fungiboletus) [inline log/controller.log:156]: I think we can ignore the issue now, but the log folder should probably not be pushed to git.

File: src/controller.ts [hunk 5]
```diff
@@ -69,7 +69,7 @@ function init(step:types.Step):void {
 let timer: SetIntervalAsyncTimer;
 
 // eslint-disable-next-line unicorn/prevent-abbreviations
-async function startContainer(image:string, stepId:number, env?: [string]) : Promise<number> {
+async function startContainer(image:string, stepId:number, env: string[]) : Promise<number> {
```
Comment (fungiboletus) [inline src/controller.ts:72]: I think I changed this line yesterday. Did you want to revert my change, or was it lost in the merge ?

File: src/server/functions.ts [hunk 6]
```diff
@@ -40,12 +40,39 @@ export async function createSimulation(model_id:string, name:string):Promise<str
...
+      'MQTT_PASS=d47AcL0|_|D1sTh3B3St',
+      'MQTT_PORT=1883'],
+  }, {
+    name: '02-storage-and-analytics',
+    step_number: 2,
+    image: 'registry.sintef.cloud/tellucare-api:latest',
+    // image: 'i1',
+    env: ['RABBITMQ_HOST=oslo.sct.sintef.no:5672',
+      'RABBITMQ_USERNAME=tellucareapi',
+      'RABBITMQ_PASSWORD=d47AcL0|_|D1sTh3B3St',
```
Comment (fungiboletus) [inline src/server/functions.ts:64]: We should probably not push hardcoded passwords.

File: src/server/functions.ts [hunk 7]
```diff
@@ -40,12 +40,39 @@ export async function createSimulation(model_id:string, name:string):Promise<str
...
+      'FHIR_URL=https://tellucloud-fhir.sintef.cloud'],
+  }, {
+    name: '03-application-logic',
+    step_number: 3,
+    image: 'registry.sintef.cloud/tellucare-application-logic:latest',
+    // image: 'i1',
+    env: ['FHIR_URL=https://tellucloud-fhir.sintef.cloud/',
+      'RABBITMQ_HOST=oslo.sct.sintef.no:5672',
+      'RABBITMQ_USERNAME=tellucareapplicationlogic',
+      'RABBITMQ_PASSWORD=d47AcL0|_|D1sTh3B3St'],
```
Comment (fungiboletus) [inline src/server/functions.ts:74]: Same there.

**Review by fungiboletus — APPROVED** [review]
Hei, I think this is fine since we are running out of time but please read my comments for the next time.

I will also see with @nvnikolov about the passwords.

**Comment by fungiboletus** [comment]
Should we get this one merged ?

**Comment by fungiboletus** [comment]
Was supersed by #9

**Comment by AleenaThomas-gh** [comment]
> Should we get this one merged ?

Yes, I think that would be better. I will create a separate branch to continue development.

**Pull request merged**